docs(backup): remote backup data might be deleted (#1004)
longhorn/website: content/docs/1.6.4/deploy/important-notes/index.md

However, Longhorn still allows the listed actions when cloning or restoring volumes created with earlier Longhorn
versions.

> ### Backup Related Custom Resources In The Cluster And Backup Data On The Remote Backup Server Might Be Deleted

Reviewer: We can remove `Backup Related Custom Resources In The Cluste` from the title and only focus on the `Backup Data On The Remote Backup Server` instead. The risk is the deletion of the backup data.

```suggestion
### Backup Related Custom Resources In The Cluster And Backup Data On The Remote Backup Server Might Be Deleted
```

Author: Modified. Thanks.

> Longhorn will attempt to clean up the backup-related custom resources (BackupVolume, BackupBackingImage, SystemBackup, and Backup) in the following scenarios:

Reviewer: Same. Need to emphasize the backup data will be deleted after Longhorn cleaned up the backup-related custom resources

Author: Modified. Thanks.

> Longhorn will attempt to delete backup data on the remote backup server and clean up the backup-related custom resources (BackupVolume, BackupBackingImage, SystemBackup, and Backup) of the cluster in the following scenarios:

Reviewer: ```suggestion
Longhorn may unintentionally delete backup-related custom resources (such as BackupVolume, BackupBackingImage, SystemBackup, and Backup) and backup data on the remote backup server in the following scenarios:
```

Author: Modified. Thanks

- An empty response from the NFS server due to a brief server downtime.

> - There is a race condition that could potentially delete the remote backup volume and its corresponding backups when resetting the backup target within a short period.

Reviewer 2: ```suggestion
- A race condition could delete the remote backup volume and its corresponding backups when the backup target is reset within a short period.
```

Author: Modified. Thanks

- An empty response from the NFS server due to a brief server downtime.
- A race condition between related Longhorn backup controllers.
- There is a race condition that could potentially delete the remote backup volume and its corresponding backups when resetting the backup target within a short period.

> The backup information will be resynchronized during the next polling interval.  

Reviewer: Need to briefly mention
- How we fix the issue and mention the ground truth is the backup data in the backup target
- After the fix, what users will see
  - backup, backup volume, ... will be deleted without deleting the backup data 
  - After the backup target is backup, the resources will be recreated

Author: Added. Thanks.

Author: Is this paragraph insufficient?

```txt
Starting with Longhorn v{{< current-version >}}:

- Longhorn does not delete the remote backup data during the synchronization.
- If the backup-related custom resources in the cluster are deleted unintentionally, they will be resynchronized from the remote backup server during the next polling interval.
```

Reviewer: > Longhorn does not delete the remote backup data during the synchronization.

The statement is not clear, and users might misunderstand that Longhorn won't delete backup data anymore. `does not delete the remote backup data` happens when satisfying some conditions. As I mentioned in `How we fix the issue and mention the ground truth is the backup data in the backup target`, Need to clearly mention the conditions. 

> After the backup target is backup, the resources will be recreated

Then, it is the second part. `After the fix, what users will see....`

Reviewer: > Is this paragraph insufficient?

No, need to update the paragraph which is confusing.

Author: Modified. Thanks

> From Longhorn v{{< current-version >}}, the remote backup data will not be deleted with the synchronization and backup information will be resynchronized during the next polling interval.  

Reviewer 2: ```suggestion
Starting with v{{< current-version >}}, Longhorn does not delete the remote backup data during synchronization. Instead, the backup data is resynchronized during the next polling period.
```

Reviewer 2: The draft is unclear so I broke up the sentence where it makes sense. Let me know if I misunderstood.

Author: Modified, and please review it again. Thanks.

However, Longhorn still allows the listed actions when cloning or restoring volumes created with earlier Longhorn
versions.

### Backup Data On The Remote Backup Server Might Be Deleted

Longhorn may unintentionally delete backup-related custom resources (such as `BackupVolume`, `BackupBackingImage`, `SystemBackup`, and `Backup`) and backup data on the remote backup server before Longhorn v{{< current-version >}} in the following scenarios:

- An empty response from the NFS server due to a brief server downtime.
- A race condition could delete the remote backup volume and its corresponding backups when the backup target is reset within a short period.

> Starting with Longhorn v{{< current-version >}}:
> 
> - Longhorn only deletes the backup-related custom resources in the cluster and does not delete anything on the remote backup server if there is a discrepancy between the backup information in the cluster and that on the remote backup server.
> - The backup-related custom resources in the cluster may be deleted unintentionally while the remote backup data remains safely stored. These resourcres will be resynchronized from the remote backup server during the next polling interval once the backup target is available.

Reviewer 2: ```suggestion
Starting with v{{< current-version >}}, Longhorn handles backup-related custom resources in the following manner:

- If there are discrepancies between the backup information in the cluster and on the remote backup server, Longhorn deletes only the backup-related custom resources in the cluster. 
- The backup-related custom resources in the cluster may be deleted unintentionally while the remote backup data remains safely stored. The deleted resources are resynchronized from the remote backup server during the next polling period (if the backup target is available).
```

Reviewer 2: I replaced "interval" with "period" because that seems to be the appropriate word for your use case. "interval" refers to the time between polling activities.

Author: Modified. Thanks.

> - After the fix, the backup-related custom resources in the cluster may be deleted unintentionally while the remote backup data remains safely stored. These resourcres will be resynchronized from the remote backup server during the next polling interval once the backup target is available.

Reviewer: ```suggestion
- The backup-related custom resources in the cluster may be deleted unintentionally while the remote backup data remains safely stored. These resourcres will be resynchronized from the remote backup server during the next polling interval once the backup target is available.
```

Author: Modified. Thanks.

However, Longhorn still allows the listed actions when cloning or restoring volumes created with earlier Longhorn
versions.

### Backup Data On The Remote Backup Server Might Be Deleted

Longhorn may unintentionally delete backup-related custom resources (such as `BackupVolume`, `BackupBackingImage`, `SystemBackup`, and `Backup`) and backup data on the remote backup server before Longhorn v{{< current-version >}} in the following scenarios:

- An empty response from the NFS server due to a brief server downtime.
- A race condition could delete the remote backup volume and its corresponding backups when the backup target is reset within a short period.

Starting with Longhorn v{{< current-version >}}:

- Longhorn only deletes the backup-related custom resources in the cluster and does not delete anything on the remote backup server if there is a discrepancy between the backup information in the cluster and that on the remote backup server.

> - You can still delete the remote backup data by Longhorn UI or deleting the `BackupVolume` or `Backup` custom resources with the `kubectl` command manually.

Author: Removed. Thanks.
